src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -11766,94 +12012,159 @@
-            {
-                // https://github.com/dotnet/roslyn/issues/33879 Conversions: Lifted operator
-                // Should this use the updated flow type and state?  How should it compute nullability?
-                return TypeWithState.Create(node.Type, NullableFlowState.NotNull);
-            }
+            BoundExpression binary;
+            Debug.Assert(stack.Count > 0);
+
+            binary = stack.Pop();
+
+            BoundExpression? leftOperand = null;
+            Conversion leftConversion = Conversion.Identity;
 
-            // Update method based on inferred operand types: see https://github.com/dotnet/roslyn/issues/29605.
-            // Analyze operator result properly (honoring [Maybe|NotNull] and [Maybe|NotNullWhen] attribute annotations) https://github.com/dotnet/roslyn/issues/32671
-            if ((object)node.LogicalOperator != null && node.LogicalOperator.ParameterCount == 2)
+            switch (binary)
             {
-                return GetReturnTypeWithState(node.LogicalOperator);
+                case BoundBinaryOperator binOp:
+                    VisitCondition(binOp.Left);
+                    break;
+                case BoundUserDefinedConditionalLogicalOperator udBinOp:
+                    (leftOperand, leftConversion) = RemoveConversion(udBinOp.Left, includeExplicitConversions: false);
+                    Visit(leftOperand);
+                    break;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(binary.Kind);
             }
-            else
+
+            while (true)
             {
-                return default;
+                switch (binary)
+                {
+                    case BoundBinaryOperator binOp:
+                        AfterLeftChildOfBinaryConditionalLogicalOperatorHasBeenVisited(binOp);
+                        break;
+                    case BoundUserDefinedConditionalLogicalOperator udBinOp:
+                        Debug.Assert(leftOperand is not null);
+                        AfterLeftChildOfBinaryConditionalLogicalOperatorHasBeenVisited(udBinOp, leftOperand, leftConversion);
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(binary.Kind);
+                }
+
+                if (stack.Count == 0)
+                {
+                    break;
+                }
+
+                AdjustConditionalState(binary);
+
+                leftOperand = binary;
+                leftConversion = Conversion.Identity;
+
+                binary = stack.Pop();
             }
         }
 
-        protected override void AfterLeftChildOfBinaryLogicalOperatorHasBeenVisited(BoundExpression node, BoundExpression right, bool isAnd, bool isBool, ref LocalState leftTrue, ref LocalState leftFalse)
+        private static void GetBinaryConditionalOperatorInfo(BinaryOperatorKind kind, out bool isAnd, out bool isBool)
```

Comment:
>GetBinaryConditionalOperatorInfo [](http://example.com/codeflow?start=28&length=32)

nit: consider bringing GetBinaryConditionalOperatorInfo and both AfterLeftChildOfBinaryConditionalLogicalOperatorHasBeenVisited methods into VisitBinaryLogicalOperatorChildren as local functions, since only used there